src/Libraries/Microsoft.Extensions.Http.Telemetry/Logging/IHttpClientLogEnricher.cs

```diff
@@ -17,5 +18,8 @@ public interface IHttpClientLogEnricher
     /// <param name="enrichmentBag">Property bag to add enriched properties to.</param>
     /// <param name="request"><see cref="HttpRequestMessage"/> object associated with the outgoing HTTP request.</param>
     /// <param name="response"><see cref="HttpResponseMessage"/> object associated with the outgoing HTTP request.</param>
-    void Enrich(IEnrichmentPropertyBag enrichmentBag, HttpRequestMessage? request = null, HttpResponseMessage? response = null);
+    /// <param name="exception">An optional <see cref="Exception"/> that was thrown within the outgoing HTTP request processing.</param>
+    // TODO: adding a new parameter to the interface method is a breaking change. Pay additional attention to this during review.
+    // Alternatively, we can decide to add another overload instead.
+    void Enrich(IEnrichmentPropertyBag enrichmentBag, HttpRequestMessage request, HttpResponseMessage? response = null, Exception? exception = null);
```

Comment:
We need to discuss this one

Reply:
No, the breaking change here is that I added `Exception? exception` - which means that existing users of this API would need to add the new parameter into their implementation of the interface